src/qforte/utils/qubit_tapering.py

```diff
@@ -0,0 +1,345 @@
+            new_basis_vector = prt_chck_mtrx_aug[row:, column].reshape((1,2 * n_qubits))
+            if dim is False:
+                basis = new_basis_vector
+                dim = True
+            else:
+                basis = np.concatenate((basis, new_basis_vector), axis=0)
+
+    # 3) Find maximal Abelian subgroup of the basis of ker{E}
+    # This is done using the symplectic bilinear form B that the binary vector space is equipped with:
+    # B(v,w) = v * B * w.T,
+    # where v and w are binary vectors and B is the matrix of the bilinear form.
+    # B(v,w) = 0 means that v and w commute while B(v,w) = 1 means that they anticommute.
+
+    ## Construct matrix of bilinear form.
+
+    zero_matrix = np.zeros((n_qubits, n_qubits), dtype=np.uint8)
+    identity_matrix = np.identity(n_qubits, dtype=np.uint8)
+    blnr_frm_mtrx = np.block([
+        [zero_matrix    , identity_matrix],
+        [identity_matrix, zero_matrix    ]
+        ])
+
+    ## Construct abelian matrix.
+    ## abelian[i,j] contains the value of the bilinear form betweeen the ith and jth basis vectors of ker{E}
+    ## 0/1 in the ith row and jth column of abelian means that the Pauli strings
+    ## associated with the ith and jth binary vectors commute/anticommute.
+
+    abelian = np.matmul(basis, np.matmul(blnr_frm_mtrx, basis.T))
+    abelian %= 2
+
+    ## Find the maximal abelian subgroup iteratively. This is done as follows:
+    ## 1) Find non-zero row of abelian with smallest Hamming weight.
+    ##    The corresponding basis vector e commutes with most other basis vectors.
+    ## 2) Find basis vectors that do not commute with e and remove them.
+    ## 3) Reconstruct abelian matrix with remaining basis vectors.
+    ## 4) If abelian == 0 exit otherwise go to step 1) and repeat.
+
+    basis_tmp = basis
+
+    while not np.all(abelian == 0):
+
+        ### Maximum possible Hamming weight of a given binary vector
+        hmng = 2 * n_qubits
+
+        idx = 0
+        for row in range(abelian.shape[0]):
+            ### Find the basis vector e that commutes with most of the remaining basis vectors
+            hmng_tmp = np.sum(abelian[row, :])
+            if hmng_tmp > 0 and hmng > hmng_tmp:
+                hmng = hmng_tmp
+                idx = row
+
+        ### Identify basis vectors that do not commute with e and eliminate them.
+        nonzero_indices = np.argwhere(abelian[idx,:])
+        basis_tmp = np.delete(basis_tmp, nonzero_indices, axis=0)
+
+        ### Construct abelian matrix for the next iteration.
+
+        abelian = np.matmul(basis_tmp, np.matmul(blnr_frm_mtrx, basis_tmp.T))
+        abelian %= 2
+
+    generators_binary = basis_tmp
+
+    ### This step is not necessary. It transforms the generators of the symmetry group in such a way that the
+    ### removed qubits have the smallest/largest possbile identity numbers. A similar modification is made
+    ### at the "sigma_x selection" part of the algorithm below.
```

Comment:
```suggestion
    ## Transform the generators to control which qubit we remove.
```

Original comment was verbose.